twyla/service/test/integration/test_event_bus.py

```diff
@@ -82,3 +86,33 @@ def test_emit_listen_roundtrip(self):
         received2 = helpers.aio_run(received[1].payload()).dict()
         del received2['meta']
         assert event_payload2 == received2
+
+
+    @mock.patch('twyla.service.events.queues')
+    def test_cancel_on_disconnect(self, mock_queues):
+        # 'mock' queue manager with an instance that we control to test the
+        # disconnect callback more easily.
+        qm = queues.QueueManager()
+        mock_queues.QueueManager.return_value = qm
+
+        async def consumer(event_name):
+            async for e in events.listen(event_name):
+                await e.ack()
+
+        async def stopper():
+            while qm.protocol is None:
+                print('wainting!')
```

Comment:
Extra `n`

Reply:
Also not sure if we want prints in here, or if we can just delete them later?

Reply:
I think that should go away.. a leftover from development.